pkg/auth/api/api.go

```diff
@@ -252,201 +201,184 @@ func (s *authService) RefreshToken(
 	if err := validateRefreshTokenRequest(req, localizer); err != nil {
 		return nil, err
 	}
-	authToken, err := s.oidc.RefreshToken(ctx, req.RefreshToken, s.opts.refreshTokenTTL, req.RedirectUrl)
+	refreshToken, err := s.verifier.VerifyRefreshToken(req.RefreshToken)
 	if err != nil {
-		if err == oidc.ErrUnregisteredRedirectURL {
-			dt, err := statusUnregisteredRedirectURL.WithDetails(&errdetails.LocalizedMessage{
-				Locale:  localizer.GetLocale(),
-				Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "redirect_url"),
-			})
-			if err != nil {
-				return nil, statusInternal.Err()
-			}
-			return nil, dt.Err()
-		}
-		if err == oidc.ErrBadRequest {
-			dt, err := statusInvalidRefreshToken.WithDetails(&errdetails.LocalizedMessage{
-				Locale:  localizer.GetLocale(),
-				Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "refresh_token"),
-			})
-			if err != nil {
-				return nil, statusInternal.Err()
-			}
-			return nil, dt.Err()
-		}
-		s.logger.Error(
-			"Failed to refresh token",
-			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
-		)
-		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+		s.logger.Error("Refresh token is invalid", zap.Any("refresh_token", refreshToken))
+		dt, err := auth.StatusUnauthenticated.WithDetails(&errdetails.LocalizedMessage{
```

Comment:
Because the refresh token is generated by the Bucketeer, we don't need to validate the `redirectURL` and `authType`.

If it fails when verifying, it will return an `Unauthenticated` error.